src/utils/common.ts

```diff
@@ -129,10 +137,133 @@ export const dirtyFieldsToRequestObject = (dirtyFields: object, form: IAppForms)
 
   syncPlanningDates(request, form);
   syncConstructionDates(request, form);
+  syncFinances(request, project);
 
   return request;
 };
 
+/**
+ * Make sure the finances are moved to correct years if dates are changed
+ * @param project IProject being updated
+ * @param data IProjectRequest object with data to be patched
+ */
+export const syncFinances = (request: IProjectRequest, project: IProject | null) => {
+  if (!project?.finances) {
+    return;
+  }
+
+  const adjustBudgets = (
+    budgetObject: IProjectFinances,
+    projectTimelineDates: IProjectEstDates,
+  ): IProjectFinances => {
+    /**
+     * Moves the provided finance value to its new finance location and sets current to null
+     * @param newFieldYear year for the new finance field which needs to be updated
+     * @param financeValue value which will be summed in the new year
+     * @param currentFinanceFieldName finance field name which the value belongs to
+     */
+    const updateFinanceValue = (
+      newFieldYear: number,
+      financeValue: number,
+      currentFinanceFieldName: keyof IProjectFinances,
+    ) => {
+      const newFinanceFieldName = yearToBudgetKey[newFieldYear - adjustedBudgetObject.year];
+      (adjustedBudgetObject[newFinanceFieldName] as string) = String(
+        safeParseInt(adjustedBudgetObject[newFinanceFieldName]) + financeValue,
+      );
+      (adjustedBudgetObject[currentFinanceFieldName] as null) = null;
+    };
+    // Year number to finance field name mapping
+    const yearToBudgetKey: { [key: number]: keyof IProjectFinances } = {
+      0: 'budgetProposalCurrentYearPlus0',
+      1: 'budgetProposalCurrentYearPlus1',
+      2: 'budgetProposalCurrentYearPlus2',
+      3: 'preliminaryCurrentYearPlus3',
+      4: 'preliminaryCurrentYearPlus4',
+      5: 'preliminaryCurrentYearPlus5',
+      6: 'preliminaryCurrentYearPlus6',
+      7: 'preliminaryCurrentYearPlus7',
+      8: 'preliminaryCurrentYearPlus8',
+      9: 'preliminaryCurrentYearPlus9',
+      10: 'preliminaryCurrentYearPlus10',
+    };
+    const { estPlanningStart, estPlanningEnd, estConstructionStart, estConstructionEnd } =
+      projectTimelineDates;
+    const planningStartYear = estPlanningStart ? getYear(estPlanningStart) : null;
+    const planningEndYear = estPlanningEnd ? getYear(estPlanningEnd) : null;
+    const constructionStartYear = estConstructionStart ? getYear(estConstructionStart) : null;
+    const constructionEndYear = estConstructionEnd ? getYear(estConstructionEnd) : null;
+
+    const adjustedBudgetObject: IProjectFinances = {
+      ...budgetObject,
+    };
+    // looping over all budgetfields in the project
+    let fieldName: keyof IProjectFinances;
+    for (fieldName in adjustedBudgetObject) {
+      if (fieldName !== 'year') {
+        // current budget field year, extract from field name
```

Comment:
current budget field year is year itself in adjustedBudgetObject. Here logic tries to read years in future. Write a better comment here, please